opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java

```diff
@@ -18,54 +18,219 @@
+                        "trond", "Fixed lint warnings\n", true),
+                new HistoryEntry("1.1", dateFormat.parse("Tue Aug 12 22:09:23 2008"),
+                        "trond", "date and time created 08/08/12 22:09:23 by trond\n\n", true));
+        History expectedHistory = new History(entries);
+        assertEquals(expectedHistory, history);
+    }
+
+    /**
+     * Negative test of {@link SCCSRepository#determineParent(CommandTimeoutType)}.
+     */
+    @Test
+    void testDetermineParentInvalid() throws Exception {
+        File codemgrDir = new File(repositoryRoot, SCCSRepository.CODEMGR_WSDATA);
+        if (!codemgrDir.isDirectory()) {
+            assertTrue(codemgrDir.mkdirs());
+        }
+        File parentFile = new File(codemgrDir, "parent");
+        assertTrue(parentFile.createNewFile());
+        try (PrintWriter out = new PrintWriter(parentFile.toString())) {
+            out.println("foo");
+        }
+        assertNull(sccsRepository.determineParent(CommandTimeoutType.INDEXER));
+        assertTrue(parentFile.delete());
+    }
+
+    /**
+     * Test of {@link SCCSRepository#determineParent(CommandTimeoutType)}.
+     */
+    @Test
+    void testDetermineParent() throws Exception {
+        File codemgrDir = new File(repositoryRoot, SCCSRepository.CODEMGR_WSDATA);
+        if (!codemgrDir.isDirectory()) {
+            assertTrue(codemgrDir.mkdirs());
+        }
+        File parentFile = new File(codemgrDir, "parent");
+        assertTrue(parentFile.createNewFile());
+        final String expectedParent = "/foo";
+        try (PrintWriter out = new PrintWriter(parentFile.toString())) {
+            out.println("VERSION 1");
+            out.println(expectedParent);
+        }
+        assertEquals(expectedParent, sccsRepository.determineParent(CommandTimeoutType.INDEXER));
+        assertTrue(parentFile.delete());
+    }
+
+    private static class GetHistoryTestParams {
+        private final String revision;
+        private final boolean shouldContain;
+
+        GetHistoryTestParams(String revision, boolean shouldContain) {
+            this.revision = revision;
+            this.shouldContain = shouldContain;
+        }
+    }
+
+    private static List<GetHistoryTestParams> getHistoryGetParams() {
+        return List.of(new GetHistoryTestParams("1.1", false),
+                new GetHistoryTestParams("1.2", true));
+    }
+
+    /**
+     * Test of {@link SCCSRepository#getHistoryGet(OutputStream, String, String, String)}.
+     */
+    @ParameterizedTest
+    @MethodSource("getHistoryGetParams")
+    void testGetHistoryGet(final GetHistoryTestParams testParams) throws Exception {
+        try (InputStream inputStream = sccsRepository.getHistoryGet(repositoryRoot.toString(),
+                "main.c", testParams.revision)) {
+            assertNotNull(inputStream);
+            byte[] buffer = new byte[1024];
+            IOUtils.readFully(inputStream, buffer);
```

Comment:
Maybe https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html#readAllBytes() would be better?

Reply:
Definitely, thanks !